tests/util/src/opentelemetry/test/__init__.py

```diff
@@ -0,0 +1,47 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from traceback import format_tb
+
+
+class AssertNotRaisesMixin:
```

Comment:
That looks like a nicer implementation, will look into it :v: 

I prefer to have `assertNotRaises` as an attribute of `self`, for consistency. That is why I used a mixin.

I'm marking this as draft because I want to make this implementation completely equivalent to [`assertRaises`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises). To do that we should support a callable as well.

@lzchen [here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-api/tests/test_implementation.py#L57) you can find one test case that would be rewritten with this context.